### ISSUE TITLE
Interactive upgrade mode

### ISSUE THREAD

**Issue by quantizor** [npm/npm#18067]
#### What's the feature?
An enhanced UX for viewing available package updates, selecting one or more, and automatically applying them.

#### What problem is the feature intended to solve?
`npm outdated` has been improved much it seems, but it's still a bit of a hassle to copy and paste each of the packages you wish to update. A checkbox-style interface for choosing and applying the updates would be highly preferable.

#### Is the absence of this feature blocking you or your team? If so, how?
Not a blocker, but I definitely am slower to upgrade packages because it's somewhat painful if there's more than one or two.

#### Is this feature similar to an existing feature in another tool?
In Yarn: https://yarnpkg.com/lang/en/docs/cli/upgrade-interactive/

#### Is this a feature you're prepared to implement, with support from the npm CLI team?
Wouldn't know where to start, but always willing to give it a try.

**Comment by 0x1ad2** [comment]
You might be interested in @dylang package [npm-check](https://github.com/dylang/npm-check)

It does have an interactive mode, you can use it by passing through `-u` flag

![image](https://user-images.githubusercontent.com/1705904/34114427-0ddf8732-e413-11e7-846e-b9d8da96e9d6.png)

**Comment by quantizor** [comment]
Yeah I use it often. Just wish the functionality was built-in like yarn.

**Comment by damianobarbati** [comment]
+1 for this. This is a very important feature, why not having this into core npm features?

**Comment by legodude17** [comment]
This would make a great RFC over at https://github.com/npm/rfcs!

**Comment by alexandr2110pro** [comment]
+ 999

Like `yarn upgrade-interactive`

To be honest, this is one of the main reasons, why my team is still on yarn.. 😃

**Comment by trevordmiller** [comment]
Agreed, this would be very useful. I’m currently relying on `npm-check -u` instead of `npm outdated` + `npm upgrade`
